test/e2e/service.go

```diff
@@ -72,9 +72,19 @@ var _ = framework.KubeDescribe("Services", func() {
 	f := framework.NewDefaultFramework("services")
 
 	var c *client.Client
+	var gceController *GCEIngressController
 
 	BeforeEach(func() {
 		c = f.Client
+		By("Initializing gce controller")
+		gceController = &GCEIngressController{ns: f.Namespace.Name, Project: framework.TestContext.CloudConfig.ProjectID, c: c}
+		gceController.init()
+	})
+
+	// Platform specific cleanup
+	AfterEach(func() {
+		By("Cleaning up cloud resources")
+		cleanupGCE(gceController)
```

Comment:
this is not what i meant. I meant write an equivalent cleanup method that deletes the resources created for service.type=lb. The leaked resources are NOT from GCLB but from the service controller/cloudprovider/test failing etc. The cleanupGCE routing would invoke: 
```shell
for r in target-pools firewall-rules forwarding-rules ...; do 
  gcloud compute $r delete
done
```
in go with logging/retries/error surfacing etc, and only do this for resources associated with Services created in its own namespace, because these tests run in parallel and all share the same gce project.